tools/src/main/java/org/apache/kafka/trogdor/rest/JsonRestServer.java

```diff
@@ -101,10 +102,9 @@ public void start(Object... resources) {
         context.addServlet(servletHolder, "/*");
 
         RequestLogHandler requestLogHandler = new RequestLogHandler();
-        // Use fully qualified name to avoid deprecation warning in import statement
-        org.eclipse.jetty.server.Slf4jRequestLog requestLog = new org.eclipse.jetty.server.Slf4jRequestLog();
-        requestLog.setLoggerName(JsonRestServer.class.getCanonicalName());
-        requestLog.setLogLatency(true);
+        Slf4jRequestLogWriter slf4jRequestLogWriter = new Slf4jRequestLogWriter();
+        slf4jRequestLogWriter.setLoggerName(JsonRestServer.class.getCanonicalName());
+        CustomRequestLog requestLog = new CustomRequestLog(slf4jRequestLogWriter, CustomRequestLog.EXTENDED_NCSA_FORMAT + " %msT");
```

Comment:
How did you come up with this format string? I could be mistaken, but I think the old implementation was just `CustomRequestLog.EXTENDED_NCSA_FORMAT` without `%msT`.

Reply:
`" %msT"` is for `requestLog.setLogLatency(true);`. As you can see from the code of [AbstractNCSARequestLog](https://github.com/eclipse/jetty.project/blob/jetty-9.4.14.v20181114/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java#L205) which is extended by [Slf4jRequestLog](https://github.com/eclipse/jetty.project/blob/jetty-9.4.14.v20181114/jetty-server/src/main/java/org/eclipse/jetty/server/Slf4jRequestLog.java), it adds log latency ms to the end of the log. However, this method does not exist in `CustomRequestLog`; Instead, there is a format string `%{UNIT}T` ([documentation](http://www.eclipse.org/jetty/javadoc/9.4.15.v20190215/org/eclipse/jetty/server/CustomRequestLog.html)). Since we are using milliseconds, I added ` %msT` at the end of the log. It's why.

Reply:
Thanks, that makes sense.